addons/editor-number-arrow-keys/addon.json

```diff
@@ -18,122 +18,29 @@
     {
       "name": "Change on regular key press",
       "id": "regular",
-      "type": "select",
-      "default": "one",
-      "potentialValues": [
-        {
-          "id": "none",
-          "name": "None"
-        },
-        {
-          "id": "hundredth",
-          "name": "0.01"
-        },
-        {
-          "id": "tenth",
-          "name": "0.1"
-        },
-        {
-          "id": "one",
-          "name": "1"
-        },
-        {
-          "id": "ten",
-          "name": "10"
-        }
-      ],
-      "if": { "settings": { "useCustom": false } }
-    },
-    {
-      "name": "Change on regular key press",
-      "id": "regularCustom",
-      "type": "string",
-      "default": "1",
-      "max": 8,
-      "if": { "settings": { "useCustom": true } }
+      "type": "decimal",
+      "default": 1,
+      "min": 0,
+      "max": 100,
+      "step": "any"
```

Comment:
Currently, users can enter up to 8-digit long numbers and 6 decimal digits (7 might be possible if you use `.` instead of `0.`). Ideally, `max` should be 10,000,000 to remain backwards compatible with what we currently have. In addition, there should be some sort of decimal precision limit if we want to be consistent with what we already have, but step also adds the ability to increment by the set increment with the input's arrow buttons and that would be super slow depending on how precise we get, but maybe that doesn't matter all that much. While I value backwards compatibility, I think we could make the limits just ever so slightly stricter in a way 99.9% of people won't care about, like maybe dropping just one digit of significance.

If I were to suggest a set max and step to try and maximize flexibility while trying to not make stepping with the arrow buttons outright impractical, I'd go with `1,000,000` and `0.0001` respectively. Most people will probably select numbers within that range, right?

...Is there seriously no other way to disable stepping and the arrow buttons? When step is as low as I'm talking about, it doesn't really make sense to me.

Or we could just make incrementing numbers in block inputs not work if the setting value is too precise, with code in the userscript, but that just seems not nice. Or... we could use JS to "validate" the setting values??? If that were the case, I'd personally allow 5 or 6 decimal digits.

Reply:
At that point should the max be removed?

Reply:
> At that point should the max be removed?

Currently, the addon can't work with numbers that are too large.